File: taekwondo Spring/taekwondo/src/main/java/com/taekwondo/service/RelacionarService.java
package com.taekwondo.service;

import java.util.List;

import com.taekwondo.model.Relacionar;
import com.taekwondo.model.Evento;

public interface RelacionarService {
	public abstract List<Relacionar> getRelacionar();
	public abstract List<Evento> getRelacionadosConActividad(String actividad);
	public abstract List<String> getActividadesEvento(int id_evento);
	public abstract Relacionar getRelacionar(int id_evento, String actividad);
	public abstract void createRelacionar(Relacionar r);
	public abstract void deleteRelacionar(int id_relacionar);
	
	
}

File: proyectoTaeKwonDo/src/app/_services/relacionar.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Relacionar } from '../_models/relacionar';

@Injectable({
  providedIn: 'root'
})
export class RelacionarService {

  API_URI = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  getRelacionar(){
    return this.http.get(this.API_URI+'/relacionar');
  }
  getRelacionadosConActividad( actividad: string){
    return this.http.get(this.API_URI+'/relacionar/actividad/'+ actividad);
  }

  getActividadesEvento( id_evento: number){
    return this.http.get(this.API_URI+'/relacionar/evento/'+ id_evento);
  }

  createTener(relacionar: Relacionar){
    return this.http.post(this.API_URI+'/relacionar', relacionar);
  }

  deleteRelacionar(id_evento: number, actividad: string){
    return this.http.delete(this.API_URI+'/relacionar/'+id_evento+'/'+actividad);
  }
}
